Enable boot delegation for backward compatibility

Moving from plain Equinox in alpah2 to Karaf in beta1 is causing many problems, specifically with openHAB 1 add-ons, see e.g.
https://community.openhab.org/t/error-on-weather-binding-when-starting-openhab2/6081
https://community.openhab.org/t/avm-fritz-box-binding-oh2b1-issue/6321
https://community.openhab.org/t/tellstick-binding-for-openhab2/1582/14
https://community.openhab.org/t/we-need-your-help-on-testing/298/29

and many more. This is very annoying since all of this used to work smoothly before.

Many of the issues seem to be due to the fact that JDK classes such as javax.xml are not on the bootclasspath. I wonder if this can be changed for the sake of backward compatibility (knowing that it is no the "clean osgi way")?

To our Karaf experts @dvanherbergen, @maggu2810, @anierbeck, @cschneider: Do you have a suggestion what we should do?

Yes, it is caused by the boot delegation.
```text
2016-01-17 15:48:58.059 [ERROR] [.w.internal.bus.WeatherBinding:57   ] - Error activating WeatherBinding: unexpected element (uri:"", local:"common-id-mappings"). Expected elements are (none)javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"common-id-mappings"). Expected elements are (none)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:726)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:247)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:242)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:109)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1131)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:556)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:538)
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.SAXConnector.startElement(SAXConnector.java:153)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:509)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:380)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:614)
```

Adding the annotation package (as @cschneider already mentioned) solves this error.

Have seen another error after the first one was solved:
```text
2016-01-17 15:54:27.255 [ERROR] [.w.internal.bus.WeatherBinding:57   ] - Error activating WeatherBinding: CommonId for provider null with id 200, already existsjava.lang.IllegalArgumentException: CommonId for provider null with id 200, already exists
	at org.openhab.binding.weather.internal.parser.CommonIdHandler.addCommonId(CommonIdHandler.java:85)
	at org.openhab.binding.weather.internal.parser.CommonIdHandler.loadMapping(CommonIdHandler.java:73)
	at org.openhab.binding.weather.internal.bus.WeatherBinding.activate(WeatherBinding.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
```

But I don't think we should mix this issues here.

So at least still a missing import...

--- a/distributions/distribution-resources/src/main/resources/etc/custom.properties
+++ b/distributions/distribution-resources/src/main/resources/etc/custom.properties
@@ -1,2 +1,3 @@
 karaf.systemBundlesStartLevel=50
 karaf.framework=equinox
+osgi.compatibility.bootdelegation=true

